routes/view-user-projects.js

```diff
@@ -15,6 +15,11 @@ module.exports = function(config) {
         return;
       }
 
+      if (response.statusCode === 404) {
+        // If there aren't any projects for this user, create one with a redirect
+        res.redirect(301, "/newProject");
```

Comment:
Not true, it didn't account for it before because it was merely rendering a page (the client passed along the query string when a user selected the project), so the query string is not lost! But in this case we are redirecting in which case you are stripping out the qs.